src/query_planner/abstract_select_plan.py

```diff
@@ -0,0 +1,61 @@
+"""Abstract class for all the scan planners
```

Comment:
Confused. we already have an abstract scan then why abstract select?

Reply:
We were not sure about the naming scheme. Should we just port over the methods from abstract select to abstract scan and delete abstract select after?

Reply:
Yeah go ahead and do that.

Reply:
I have changed the references of select plan to seq scan plan. If this is correct, I will delete the abstract select plan and logical select plan.